src/topp/IDFileConverter.cpp

```diff
@@ -615,28 +617,49 @@ class TOPPIDFileConverter :
     else if (out_type == FileTypes::FASTA)
     {
       Size count = 0;
-      ofstream fasta(out.c_str(), ios::out);
+
+      bool concat = getFlag_("concatenate_peptides");
+      //To avoid creating (K/R)P sites and losing information when the FASTA entry is used by
+      //a search engine; Peptides beginning with 'P' are saved seperatly and later moved to the beginning of the concatenated sequence.
+      String all_p(""); //peptides beginning with 'P'
+      String all_but_p(""); //all the others
+
+      FASTAFile f;
+      f.writeStart(out);
+      FASTAFile::FASTAEntry entry;
       for (Size i = 0; i < peptide_identifications.size(); ++i)
       {
-        for (Size l = 0; l < peptide_identifications[i].getHits().size(); ++l)
+        if (concat)
+        {
+          if (peptide_identifications[i].getHits()[0].getSequence().toString()[0] == 'P')
+          {
+            all_p += peptide_identifications[i].getHits()[0].getSequence().toString();
```

Comment:
- do you really want the modified peptide sequence, not the unmodified one?
- calling `toString()` is expensive. Compute it once and use it for comparing to 'P'.
- you are missing an `else`...